File: src/main/kotlin/dev/slne/surf/essentials/commands/cheat/gui/SmithingTableCommand.java
package dev.slne.surf.essentials.commands.cheat.gui;

import dev.slne.surf.essentials.utils.permission.Permissions;

public class SmithingTableCommand extends GuiCommand {
    SmithingTableCommand() {
        super("smithingtable", "Der Schmiedetisch", Permissions.SMITHING_TABLE_SELF_PERMISSION, Permissions.SMITHING_TABLE_OTHER_PERMISSION, "smithing");
    }

    @Override
    public void open(org.bukkit.entity.Player player) {
        player.openSmithingTable(player.getLocation(), true);
    }
}

File: src/main/kotlin/dev/slne/surf/essentials/command/table/SmithingTableCommand.kt
@file:Suppress("DEPRECATION")

package dev.slne.surf.essentials.command.table

import dev.jorel.commandapi.kotlindsl.commandTree
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.api.core.messages.adventure.sendText
import dev.slne.surf.essentials.util.permission.EssentialsPermissionRegistry

fun smithingTableCommand() = commandTree("smithingtable") {
    withPermission(EssentialsPermissionRegistry.SMITHING_TABLE_COMMAND)
    playerExecutor { player, _ ->
        player.openSmithingTable(null, true)
        player.sendText {
            appendSuccessPrefix()
            success("Du hast einen Schmiedetisch geöffnet.")
        }
    }
}